sklearn/discriminant_analysis.py

```diff
@@ -623,11 +623,16 @@ class QuadraticDiscriminantAnalysis(BaseEstimator, ClassifierMixin):
         Discriminant Analysis
     """
 
-    def __init__(self, priors=None, reg_param=0., store_covariances=False,
-                 tol=1.0e-4):
+    def __init__(self, priors=None, reg_param=0., store_covariance=False,
+                 tol=1.0e-4, store_covariances=False):
+        if store_covariances is not None:
```

Comment:
Since `store_covariances` is deprecated in `fit` which will be removed in `0.19` therefore setting the value only through `init`, shouldn't this be done in `init`? @amueller